karapace/utils.py

```diff
@@ -73,17 +84,34 @@ def default_json_serialization(
 SEPARATORS = (",", ":")
 
 
-def json_encode(obj, *, sort_keys: bool = True, binary=False, compact=False):
+def json_encode(obj, *, sort_keys: bool = True, binary=False, compact=False, indent: Optional[int] = None):
     separators = SEPARATORS if compact else None
+    arguments = {
+        "sort_keys": sort_keys,
+        "default": default_json_serialization,
+        "separators": separators,
+    }
+    if indent is not None:
+        arguments["indent"] = indent
     res = json.dumps(
         obj,
-        sort_keys=sort_keys,
-        default=default_json_serialization,
-        separators=separators,
+        **arguments,
     )
     return res.encode("utf-8") if binary else res
 
 
+def json_decode(content: [str, IO]) -> JsonData:
+    if isinstance(content, str):
+        try:
+            return json.loads(content)
+        except JSONDecodeError as ex:
+            raise ex
+    try:
+        return json.load(content)
+    except JSONDecodeError as ex:
+        raise ex
```

Comment:
There is no need to catch and rethrow the same exception, and `bytes` is a perfectly valid input for both `json` and `usjon` as well. By including `bytes` we can also get rid of **ALL** `.decode("utf8")` that are currently sprinkled across the codebase when calling this function.

```suggestion
def json_decode(it: Union[bytes, str, IO]) -> JsonData:
    if isinstance(content, bytes) or isinstance(content, str):
        return json.loads(content)
    else:
        return json.load(content)
```

Reply:
Yes, rethrow was a workaround to flake8 and pylint unused import nag.